lib/ddtrace/contrib/rails/core_extensions.rb

```diff
@@ -90,30 +99,15 @@ def patch_partial_renderer(klass)
       do_once(:patch_partial_renderer) do
         klass.class_eval do
           def render_with_datadog(*args, &block)
-            # Create a tracing context and start the rendering span
-            tracing_context = {}
-            Datadog::Contrib::Rails::ActionView.start_render_partial(tracing_context: tracing_context)
-            tracing_contexts[current_span_id] = tracing_context
-
-            render_without_datadog(*args)
-          rescue Exception => e
-            # attach the exception to the tracing context if any
-            tracing_contexts[current_span_id][:exception] = e
-            raise e
-          ensure
-            # Ensure that the template `Span` is finished even during exceptions
-            # Remove the existing tracing context (to avoid leaks)
-            tracing_contexts.delete(current_span_id)
-
-            # Then finish the span associated with the context
-            Datadog::Contrib::Rails::ActionView.finish_render_partial(tracing_context: tracing_context)
+            datadog_tracer.trace('rails.render_partial', span_type: Datadog::Ext::HTTP::TEMPLATE) do |span|
+              with_datadog_span(span) { render_without_datadog(*args) }
+            end
           end
 
           def render_partial_with_datadog(*args)
             begin
-              # update the tracing context with computed values before the rendering
               template_name = Datadog::Contrib::Rails::Utils.normalize_template_name(@template.try('identifier'))
-              tracing_contexts[current_span_id][:template_name] = template_name
+              active_datadog_span.set_tag('rails.template_name', template_name) if template_name
```

Comment:
`'rails.template_name'.freeze` this will avoid allocating (and then GCing) the same string multiple times within single request.

Reply:
You know, I tried doing exactly this, after I saw your PR. It's actually not possible to add a constant to the module we're patching here, because it raises a dynamic constant error. The only way I could think around this is either 1) refactoring this into a module (preferable, but must defer per feedback above), or 2) defining the constant in another namespace then referencing that.

The 2nd option I could do now, it just won't be as clean.